iXBRLViewerPlugin/viewer/src/js/report.js

```diff
@@ -69,20 +69,32 @@ iXBRLReport.prototype._initialize = function () {
     }
 }
 
-iXBRLReport.prototype.isCalculationContributingConcept = function(c) {
-    if (!this.data.rels?.calc) return false;
-    return Boolean(Object.values(this.data.rels.calc).some(calculations => {
-        return Object.values(calculations).some(contributingItems => {
-            return contributingItems.some(i => i.t === c);
-        })
-    }));
-}
-
-iXBRLReport.prototype.isCalculationSummationConcept = function(c) {
-    if (!this.data.rels?.calc) return false;
-    return Boolean(Object.values(this.data.rels.calc).some(calculations => {
-        return Object.keys(calculations).includes(c);
-    }));
+iXBRLReport.prototype.isCalculationContributor = function(c) {
+    if (this._calculationContributors === undefined) {
+        if (this.data.rels?.calc) {
+            this._calculationContributors = Object.values(this.data.rels.calc).flatMap(calculations => {
```

Comment:
Suggest wrapping in  `new Set` and use `.has` below.  Otherwise, this is still O(facts*contributors), just deferred until the first use of the filter.

With that, do we need the pseudo-private fact field properties at all, or can the method just forward?  I don't know much about JavaScript, but I've read that engines like to optimize based on the shape (fields) of an object, so I don't know the effect on dynamically adding a property.